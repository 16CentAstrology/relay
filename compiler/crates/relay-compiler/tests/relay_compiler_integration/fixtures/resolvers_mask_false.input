//- User_foo.js
/**
 * @RelayResolver User.foo: String
 * @rootFragment UserFooFragment
 */
graphql`fragment UserFooFragment on User {
  ...UserBarFragment @relay(mask: false)
}`

//- User_bar.js
/**
 * @RelayResolver User.bar: String
 * @rootFragment UserBarFragment
 */
graphql`fragment UserBarFragment on User @relay(mask: false) {
  name
}`

//- relay.config.json
{
  "language": "typescript",
  "schema": "schema.graphql",
  "resolversSchemaModule": {
    "path": "__generated__/ResolversSchemaModule.js"
  }
}

//- schema.graphql
type Query { me: User }
type User { name: String }
