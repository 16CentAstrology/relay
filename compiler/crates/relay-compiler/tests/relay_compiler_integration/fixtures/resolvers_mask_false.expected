==================================== INPUT ====================================
//- User_foo.js
/**
 * @RelayResolver User.foo: String
 * @rootFragment UserFooFragment
 */
graphql`fragment UserFooFragment on User {
  ...UserBarFragment @relay(mask: false)
}`

//- User_bar.js
/**
 * @RelayResolver User.bar: String
 * @rootFragment UserBarFragment
 */
graphql`fragment UserBarFragment on User @relay(mask: false) {
  name
}`

//- relay.config.json
{
  "language": "typescript",
  "schema": "schema.graphql",
  "resolversSchemaModule": {
    "path": "__generated__/ResolversSchemaModule.js"
  }
}

//- schema.graphql
type Query { me: User }
type User { name: String }
==================================== OUTPUT ===================================
//- __generated__/ResolversSchemaModule.js
/**
 * <auto-generated> SignedSource<<bbd6a4dbf3ce5ec1cb2407e258b9ddd4>>
 * @lightSyntaxTransform
 * @nogrep
 */

/* tslint:disable */
/* eslint-disable */
// @ts-nocheck

import { SchemaResolvers } from 'ReactiveQueryExecutor';
import { ResolverFunction, NormalizationSplitOperation } from 'relay-runtime';


import {bar as userBarResolver} from '../User_bar';
import {foo as userFooResolver} from '../User_foo';
import UserBarFragment$normalization from 'UserBarFragment$normalization.graphql';
import UserFooFragment$normalization from 'UserFooFragment$normalization.graphql';

const schema_resolvers: SchemaResolvers = {
  "User": {
    "bar": {
      "resolverFunction": userBarResolver,
      "rootFragment": UserBarFragment$normalization
    },
    "foo": {
      "resolverFunction": userFooResolver,
      "rootFragment": UserFooFragment$normalization
    }
  }
};

export default schema_resolvers;

//- __generated__/UserBarFragment$normalization.graphql.ts
/**
 * <auto-generated> SignedSource<<95e38be19d7d9ab3536081c3acadc575>>
 * @lightSyntaxTransform
 * @nogrep
 */

/* tslint:disable */
/* eslint-disable */
// @ts-nocheck

import { NormalizationSplitOperation } from 'relay-runtime';


const node: NormalizationSplitOperation = {
  "kind": "SplitOperation",
  "metadata": {},
  "name": "UserBarFragment$normalization",
  "selections": [
    {
      "alias": null,
      "args": null,
      "kind": "ScalarField",
      "name": "name",
      "storageKey": null
    }
  ]
};

(node as any).hash = "5b5f550cc89cad7f59a3b8585e7c3f89";

export default node;

//- __generated__/UserBarFragment.graphql.ts
/**
 * <auto-generated> SignedSource<<b1196d270eb57cfbbe17f0ae000071c2>>
 * @lightSyntaxTransform
 * @nogrep
 */

/* tslint:disable */
/* eslint-disable */
// @ts-nocheck

import { ReaderFragment } from 'relay-runtime';
import { FragmentRefs } from "relay-runtime";
export type UserBarFragment$data = {
  readonly name: string | null | undefined;
};
export type UserBarFragment$key = {
  readonly " $data"?: UserBarFragment$data;
  readonly " $fragmentSpreads": FragmentRefs<"UserBarFragment">;
};

const node: ReaderFragment = {
  "argumentDefinitions": [],
  "kind": "Fragment",
  "metadata": {
    "mask": false
  },
  "name": "UserBarFragment",
  "selections": [
    {
      "alias": null,
      "args": null,
      "kind": "ScalarField",
      "name": "name",
      "storageKey": null
    }
  ],
  "type": "User",
  "abstractKey": null
};

(node as any).hash = "5b5f550cc89cad7f59a3b8585e7c3f89";

export default node;

//- __generated__/UserFooFragment$normalization.graphql.ts
/**
 * <auto-generated> SignedSource<<1348defc58e15858601d3e8a94291d87>>
 * @lightSyntaxTransform
 * @nogrep
 */

/* tslint:disable */
/* eslint-disable */
// @ts-nocheck

import { NormalizationSplitOperation } from 'relay-runtime';


const node: NormalizationSplitOperation = {
  "kind": "SplitOperation",
  "metadata": {},
  "name": "UserFooFragment$normalization",
  "selections": [
    {
      "alias": null,
      "args": null,
      "kind": "ScalarField",
      "name": "name",
      "storageKey": null
    }
  ]
};

(node as any).hash = "5adc58968744f5863661d6083f1e8227";

export default node;

//- __generated__/UserFooFragment.graphql.ts
/**
 * <auto-generated> SignedSource<<940f80fffbe6917e30935f952e5c589e>>
 * @lightSyntaxTransform
 * @nogrep
 */

/* tslint:disable */
/* eslint-disable */
// @ts-nocheck

import { ReaderFragment } from 'relay-runtime';
import { FragmentRefs } from "relay-runtime";
export type UserFooFragment$data = {
  readonly name: string | null | undefined;
  readonly " $fragmentType": "UserFooFragment";
};
export type UserFooFragment$key = {
  readonly " $data"?: UserFooFragment$data;
  readonly " $fragmentSpreads": FragmentRefs<"UserFooFragment">;
};

const node: ReaderFragment = {
  "argumentDefinitions": [],
  "kind": "Fragment",
  "metadata": null,
  "name": "UserFooFragment",
  "selections": [
    {
      "alias": null,
      "args": null,
      "kind": "ScalarField",
      "name": "name",
      "storageKey": null
    }
  ],
  "type": "User",
  "abstractKey": null
};

(node as any).hash = "5adc58968744f5863661d6083f1e8227";

export default node;
